src/typechecker.erl

```diff
@@ -3526,6 +3526,19 @@ add_type_pat({record, P, Record, Fields}, Ty, TEnv, VEnv) ->
             {VEnv2, Cs2} = add_type_pat_fields(Fields, Record, TEnv, VEnv),
             {type(none), Ty, VEnv2, constraints:combine(Cs1, Cs2)}
     end;
+add_type_pat({map, _, _} = MapPat, {var, _, Var} = TyVar, _TEnv, VEnv) ->
+    %% FIXME this is a quite rudimentary implementation
+    %% - variables from the map pattern become any()
+    %% - the contraint could contain the map keys
+    Cs = constraints:add_var(
```

Comment:
I think it's fine that we leave this in as a FIXME. We need to redo the constraint system anyways.